pkg/operation/botanist/component/extensions/dnsrecord/dnsrecord.go

```diff
@@ -275,3 +278,11 @@ func (c *dnsRecord) SetRecordType(recordType extensionsv1alpha1.DNSRecordType) {
 func (c *dnsRecord) SetValues(values []string) {
 	c.values.Values = values
 }
+
+func (c *dnsRecord) valuesDontMatchDNSRecord() bool {
+	return c.values.SecretName != c.dnsRecord.Spec.SecretRef.Name ||
+		!pointer.StringEqual(c.values.Zone, c.dnsRecord.Spec.Zone) ||
+		c.values.DNSName != c.dnsRecord.Spec.Name ||
```

Comment:
`Spec.Name` is also immutable.
```suggestion
```

Reply:
Ah, totally missed this. Thanks for being thorough!